manager/orchestrator/taskreaper/task_reaper_test.go

```diff
@@ -0,0 +1,165 @@
+package taskreaper
+
+import (
+	"context"
+
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/state/store"
+)
+
+// TestTaskReaperInit tests that the task reaper correctly cleans up tasks when
```

Comment:
This bug was for the service removal case. We should update that test to catch the bug.

Reply:
@anshulpundir does https://github.com/docker/swarmkit/pull/2476 addresses that? I mean that PR should only go green when this PR is merged?

Reply:
I replied back to @dperny in a separate message. This fix is correct and I realized that as I was adding the unit test in #2476 @marcusmartins